src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/DisableRuntimeMarshallingAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*=============================================================================
+**
+**
+**
+** Purpose: Mark up the program to indicate various legacy or new opt-in behaviors.
+**
+**
+=============================================================================*/
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Disables the built-in runtime managed/unmanaged marshalling subsystem for
+    /// P/Invokes, Delegate types, and unmanaged function pointer invocations.
+    /// </summary>
+    /// <remarks>
+    /// The built-in marshalling subsystem has some behaviors that cannot be changed due to
+    /// backward-compatibility requirements. This attribute allows disabling the built-in
+    /// subsystem and instead uses the following rules for P/Invokes, Delegates,
+    /// and unmanaged function pointer invocations:
+    ///
+    /// - All value types that do not contain reference type fields recursively (<c>unmanaged</c> in C#) are blittable
+    /// - Value types that recursively have any fields that are not enums and have <c>[StructLayout(LayoutKind.Auto)]</c> are disallowed from interop.
+    /// - All reference types are disallowed from usage in interop scenarios.
+    /// - SetLastError support in P/Invokes is disabled.
+    /// - varargs support is disabled.
+    /// - LCIDConversionAttribute support is disabled.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false, AllowMultiple = false)]
+    public sealed class DisableRuntimeMarshallingAttribute : Attribute
```

Comment:
Have you tried to mark CoreLib with this and see what breaks? Should we add it to CoreLib as part to this PR?

Reply:
I haven't tried applying this attribute to CoreLib yet. It looks like we have a few regular `DllImport`s that are waiting on the next phase of work after this attribute gets in (described in #63765) to handle cases where we pass `System.Guid`s to and from native code. Once we put this PR in, we can update the DllImportGenerator based on #63765, convert the rest of the DllImports over to the generated ones, and apply this attribute.

I might throw up another PR next week based on this one that applies the attribute just to see all that breaks.